src/replication.c

```diff
@@ -3684,13 +3684,18 @@ void syncWithPrimary(connection *conn) {
 
     /* Prepare a suitable temp file for bulk transfer */
     if (!useDisklessLoad()) {
+        /* We save the errno of open to prevent some systems from modifying it after
+         * the sleep call. For example, sleep in Mac will change errno to ETIMEDOUT. */
+        int saved_errno;
         while (maxtries--) {
             snprintf(tmpfile, 256, "temp-%d.%ld.rdb", (int)server.unixtime, (long int)getpid());
             dfd = open(tmpfile, O_CREAT | O_WRONLY | O_EXCL, 0644);
             if (dfd != -1) break;
```

Comment:
another choice is that we can print the error message in each try, so we don't need this saved_errno. The cost is that we may print 5 times of the log.

Reply:
yes, it is no needed